pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, Optional, Sized, cast
+
+import torch
+from torch.utils.data import DataLoader, Dataset, Sampler
+
+from pystiche.data import ImageFolderDataset
+from pystiche.image.transforms import ComposedTransform, Transform, ValidRandomCrop
+from pystiche.image.transforms import functional as F
+from pystiche.image.utils import extract_image_size
+
+from ..data.utils import FiniteCycleBatchSampler
+from ..utils import OptionalGrayscaleToFakegrayscale
+
+
+def image_transform(edge_size: int = 768,) -> ComposedTransform:
+    class OptionalRescale(Transform):
+        # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/prepare_dataset.py#L49-L68
+        def __init__(
+            self, interpolation_mode: str = "bilinear",
+        ):
+            super().__init__()
+            self.interpolation_mode = interpolation_mode
+
+        def forward(
+            self, image: torch.Tensor, maximal_size: int = 1800, minimal_size: int = 800
+        ) -> torch.Tensor:
+            if max(image.shape) > maximal_size:
```

Comment:
Please correct me if I'm wrong:

- If the image is larger than `maximal_size` or smaller than `minimal_size`, it is resized to that.
- Special case: if the image is smaller than four times the `minimal_size` it is resized to the `minimal_size` on both edges

Just making sure, that I understand it correctly before reviewing.

Reply:
Yeah, that's right.

Reply:
In that case, why are you using `Rescale` at all? I think `Resize` would be the better fit here. As a rule of thumb: If you the size you want to resize to, use `Resize`. If the size you want to resize to is defined relative to the original size, i.e. double or half, use `Rescale`.